src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -44,6 +54,62 @@ def infer_schema_from_description(self, table_or_query: str) -> StructType:
             cursor.close()
             conn.close()
 
+    def udtf_ingestion(
+        self,
+        session: "snowflake.snowpark.Session",
+        schema: StructType,
+        partition_table: str,
+        external_access_integrations: str,
+        fetch_size: int = 1000,
+        imports: Optional[List[str]] = None,
+        packages: Optional[List[str]] = None,
+    ) -> "snowflake.snowpark.DataFrame":
+        from snowflake.snowpark._internal.data_source.utils import UDTF_PACKAGE_MAP
+
+        udtf_name = f"data_source_udtf_{generate_random_alphanumeric(5)}"
+        session.udtf.register(
+            self.udtf_class_builder(fetch_size=fetch_size),
+            name=udtf_name,
+            output_schema=StructType(
+                [
+                    StructField(field.name, StringType(), field.nullable)
+                    for field in schema.fields
+                ]
+            ),
+            external_access_integrations=[external_access_integrations],
+            packages=packages or UDTF_PACKAGE_MAP.get(self.dbms_type),
+            imports=imports,
+        )
+        call_udtf_sql = f"""
+            select * from {partition_table}, table({udtf_name}({PARTITION_TABLE_COLUMN_NAME}))
+            """
+        start = time.time()
+        res = session.sql(call_udtf_sql)
+        logger.info(
+            f"ingest data to snowflake udtf takes: {time.time() - start} seconds"
+        )
+        cols = [
+            res[field.name].cast(field.datatype).alias(field.name)
+            for field in schema.fields
+        ]
+        return res.select(cols)
+
+    def udtf_class_builder(self, fetch_size: int = 1000) -> type:
+        create_connection = self.create_connection
+
+        class UDTFIngestion:
+            def process(self, query: str):
+                conn = create_connection()
+                cursor = conn.cursor()
+                cursor.execute(query)
+                while True:
+                    rows = cursor.fetchmany(fetch_size)
+                    if not rows:
+                        break
+                    yield from rows
```

Comment:
I see this part of logic repeated a couple of time in each driver implementation, can we make it a reusable function?